retis/src/process/cli/pcap.rs

```diff
@@ -68,76 +77,118 @@ macro_rules! some_or_return {
-        // Having a common & a kernel section is mandatory for now, seeing a
-        // filtered event w/o one of those is bogus.
-        let common = event
-            .common
-            .as_ref()
-            .ok_or_else(|| anyhow!("No common section in event"))?;
-        let kernel = event
-            .kernel
-            .as_ref()
-            .ok_or_else(|| anyhow!("No skb section in event"))?;
-
-        self.stats.processed += 1;
-
-        // The packet section is mandatory for us to generate PCAP events, but
-        // it might not be present in some filtered events. Stats are kept to
-        // inform the user.
-        let packet = some_or_return!(&event.packet, self.stats.missing_packet);
-
-        // The dev & ns sections are best to have but not mandatory to generate
-        // an event. If not found, fake them.
-        let (ifindex, ifname) = match &event.dev {
-            Some(dev) => (dev.ifindex, dev.name.as_str()),
-            None => {
-                self.stats.missing_dev += 1;
-                (0, "?")
-            }
-        };
-        let (netns_cookie, netns_inum) = match &event.netns {
-            Some(ns) => (ns.cookie, ns.inum),
-            None => {
-                self.stats.missing_ns += 1;
-                (None, 0)
+    /// Extract interface information from the event based on configuration,
+    /// adding any necessary block and updating internal cache accordingly.
+    fn process_interface(&mut self, event: &Event, blocks: &mut Vec<Block<'_>>) -> Result<u32> {
+        let iface: String;
+        let key: String;
+        let desc: String;
+
+        if self.probe_as_iface {
+            iface = if let Some(kernel) = &event.kernel {
+                format!("{}/{}", kernel.probe_type, kernel.symbol)
+            } else if let Some(user) = &event.userspace {
+                format!("{}/{}", user.probe_type, user.symbol)
+            } else {
+                bail!("no kernel or userspace sections");
+            };
+            key = iface.clone();
+            desc = format!("Fake interface for probe {}", iface);
+        } else {
+            let (ifindex, ifname) = match &event.dev {
+                Some(dev) => {
+                    desc = format!("ifindex={}", dev.ifindex);
+                    (dev.ifindex, dev.name.as_str())
+                }
+                None => {
+                    desc = "Fake Interface".to_string();
+                    self.stats.missing_dev += 1;
+                    (0, "?")
+                }
+            };
+            match &event.netns {
+                Some(ns) => {
+                    if let Some(cookie) = ns.cookie {
+                        iface = format!("{ifname} ns {:#x}/{}", cookie, ns.inum);
+                    } else {
+                        iface = format!("{ifname} ns {}", ns.inum);
+                    }
+                    key = format!("{}/{ifindex}", ns.inum);
```

Comment:
I think the key should use `cookie` instead of `inum` as only the former is unique.